embedded-google-pubsub/src/main/java/com/playtika/test/pubsub/PubSubResourcesGenerator.java

```diff
@@ -32,15 +30,12 @@
     private final CredentialsProvider credentialsProvider;
     private final TopicAdminClient topicAdminClient;
     private final SubscriptionAdminClient subscriptionAdminClient;
-    private final PubsubProperties properties;
-
     private final String projectId;
+    private final Collection<TopicAndSubscription> topicAndSubscriptions;
 
-    public PubSubResourcesGenerator(@Qualifier(PubsubProperties.BEAN_NAME_EMBEDDED_GOOGLE_PUBSUB) GenericContainer pubsub,
-                                    PubsubProperties properties) throws IOException {
-        this.properties = properties;
-        this.projectId = properties.getProjectId();
-        ManagedChannel channel = ManagedChannelBuilder.forAddress(properties.getHost(), pubsub.getMappedPort(properties.getPort())).usePlaintext().build();
+    public PubSubResourcesGenerator(ManagedChannel channel, String projectId, Collection<TopicAndSubscription> topicAndSubscriptions) throws IOException {
```

Comment:
This class ended with a better design, it receives exactly what it needs to do its work. Thanks @tdanylchuk for pushing me to a test :)